packages/siwe/lib/client.ts

```diff
@@ -81,6 +81,7 @@ export class SiweMessage {
       }
     }
     this.nonce = this.nonce || generateNonce();
+    this.issuedAt = this.issuedAt || new Date().toISOString();
```

Comment:
I think it would be better to keep the generation in `toMessage` to make sure it represents as best we can when it is issued. So either it's generated earlier or the validation checks whether it's null or not. And the type of `issuedAt` should represent that optionality as well.